code/game/objects/items/legcuffs.dm

```diff
@@ -92,3 +93,45 @@
 					var/mob/living/simple_animal/SA = AM
 					SA.health -= 20
 	..()
+
+
+/obj/item/restraint/legcuffs/xeno_restraints
+	name = "xeno restraints"
+	desc = "Use this to hold xenomorphic creatures moderately-safely."
+	flags_equip_slot = SLOT_WAIST
+	throwforce = 5
+	w_class = SIZE_SMALL
+	throw_range = 5
+	matter = list("metal" = 500)
+
+	breakouttime = 2 MINUTES
+
+/obj/item/restraint/legcuffs/xeno_restraints/apply_legcuffs(mob/living/carbon/target, mob/user)
+	return FALSE
+
+/obj/item/restraint/legcuffs/xeno_restraints/attack(mob/living/carbon/target_mob, mob/user as mob)
+	if(!istype(target_mob, /mob/living/carbon/xenomorph))
+		to_chat(user, SPAN_DANGER("The cuffs do not fit!"))
+		return
+	if(!target_mob.legcuffed)
+		var/turf/user_loc = user.loc
+		var/turf/target_loc = target_mob.loc
+		playsound(src.loc, 'sound/weapons/handcuffs.ogg', 25, 1, 6)
+		target_mob.visible_message(SPAN_DANGER("<B>[user] is trying to put restraints on [target_mob]!</B>"))
+		if(!do_after(user, 3 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, target_mob, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
+			return
+		if(!target_mob)
+			return
+		if(user_loc == user.loc && target_loc == target_mob.loc)
```

Comment:
```
if(user_loc != get_turf(user) || target_loc != get_turf(target_mob))
	return
```
may be better--- but I reckon just checking the distance between the user and the target would be a better solution than checking they're on the same turf specifically.